sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -0,0 +1,682 @@
+    {% endif %}
+  GROUP BY
+    client_id,
+    sample_id
+),
+main_ping AS (
+  SELECT
+    client_id AS client_id,
+    sample_id AS sample_id,
+    -- The submission_timestamp_min is used to compare with the TIMESTAMP of
+    -- the new_profile and first shutdown pings.
+    -- It was implemented on Dec 16, 2019 and has data from 2018-10-30.
+    IF(
+      ANY_VALUE(submission_date HAVING MIN submission_date) >= '2018-10-30',
+      MIN(submission_timestamp_min),
+      TIMESTAMP(ANY_VALUE(submission_date HAVING MIN submission_date))
+    ) AS first_seen_date,
+    TIMESTAMP(
+      ARRAY_AGG(submission_date IGNORE NULLS ORDER BY submission_date)[SAFE_OFFSET(1)]
+    ) AS second_seen_date,
+    CAST(NULL AS STRING) AS architecture,
+    ANY_VALUE(app_build_id HAVING MIN submission_date) AS app_build_id,
+    ANY_VALUE(app_name HAVING MIN submission_date) AS app_name,
+    CAST(NULL AS STRING) AS platform_version,
+    ANY_VALUE(vendor HAVING MIN submission_date) AS vendor,
+    ANY_VALUE(app_version HAVING MIN submission_date) AS app_version,
+    CAST(NULL AS STRING) AS xpcom_abi,
+    CAST(NULL AS STRING) AS document_id,
+    ANY_VALUE(distribution_id HAVING MIN submission_date) AS distribution_id,
+    CAST(NULL AS STRING) AS partner_distribution_version,
+    CAST(NULL AS STRING) AS partner_distributor,
+    CAST(NULL AS STRING) AS partner_distributor_channel,
+    CAST(NULL AS STRING) AS partner_id,
+    ANY_VALUE(attribution.campaign HAVING MIN submission_date) AS attribution_campaign,
+    ANY_VALUE(attribution.content HAVING MIN submission_date) AS attribution_content,
+    ANY_VALUE(attribution.experiment HAVING MIN submission_date) AS attribution_experiment,
+    ANY_VALUE(attribution.medium HAVING MIN submission_date) AS attribution_medium,
+    ANY_VALUE(attribution.source HAVING MIN submission_date) AS attribution_source,
+    CAST(NULL AS STRING) AS attribution_ua,
+    ANY_VALUE(
+      default_search_engine_data_load_path
+      HAVING
+        MIN submission_date
+    ) AS engine_data_load_path,
+    ANY_VALUE(default_search_engine_data_name HAVING MIN submission_date) AS engine_data_name,
+    ANY_VALUE(default_search_engine_data_origin HAVING MIN submission_date) AS engine_data_origin,
+    ANY_VALUE(
+      default_search_engine_data_submission_url
+      HAVING
+        MIN submission_date
+    ) AS engine_data_submission_url,
+    CAST(NULL AS STRING) AS apple_model_id,
+    ANY_VALUE(city HAVING MIN submission_date) AS city,
+    CAST(NULL AS STRING) AS db_version,
+    ANY_VALUE(geo_subdivision1 HAVING MIN submission_date) AS subdivision1,
+    ANY_VALUE(normalized_channel HAVING MIN submission_date) AS normalized_channel,
+    ANY_VALUE(country HAVING MIN submission_date) AS country,
+    ANY_VALUE(os HAVING MIN submission_date) AS normalized_os,
+    ANY_VALUE(normalized_os_version HAVING MIN submission_date) AS normalized_os_version,
+    CAST(NULL AS STRING) AS startup_profile_selection_reason,
```

Comment:
cc @gkabbz 

This is the part that we potentially need backfilled into clients_daily_v6, [pulling from this probe](https://probes.telemetry.mozilla.org/?search=reason&view=detail&probeId=scalar%2Fstartup.profile_selection_reason) in the main ping. 

This is needed so long as we want to maintain the current logic of the NtM dashboard. The dashboard only counts new profiles with reason = [`firstrun-created-default`](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/telemetry_derived/desktop_funnel_new_profiles_v1/query.sql#L17). This is to help us ID new profiles that were created from an install, rather than ones that were created for some other reason. 

Without this addition, the dashboard fka NtM will ignore new profiles in the case the first ping we receive from them is a main ping.

Reply:
that's a good point. The aggregation might be problematic. Taking the first of the day would be useful for **our** purposes, but it might not be true for any other/future use cases that need this field. Adding it to `clients_daily` might just not make sense. 

Other options to consider. Is pulling the data directly from main pings, and bypassing `clients_daily` an option, or is it way too cost prohibitive? I know the history of main pings doesn't go far back enough (we might only keep main pings 18 months or something), but can we initialize with some combination of both (use main pings where available, otherwise use `clients_daily`), and then just look at main pings for the daily updates?

Reply:
also, to add, `attribution.ua` is also a field necessary for Numbers that Matter, that's not available in `telemetry.clients_daily` (but is available in the raw ping). 

[line 289](https://github.com/mozilla/bigquery-etl/blob/dabd70ad9f3bd2bcbdfccbab29c738b529d99aa4/sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql#L289C5-L289C44)

Reply:
Yes, there are 14 columns in the query that are not available in `clients_daily_v6` but are available in the ping table `main_v4` (added a comment in the query with each corresponding column).
In the query, these are retrieved from `new_profile` and `first_shutdown` ping and would be NULL if the main ping reports the first_seen_date.
Is there any other source that could be used for the dashboard while these data is available in `clients_daily_v6`?

Reply:
Alternatively, we could retrieve startup_profile_reason and attribution.ua from another ping if it's available.

Reply:
Resolving as per [decision](https://github.com/mozilla/bigquery-etl/pull/3962#pullrequestreview-1639351082).